### DISCUSSION TITLE
How to use gitignore.LoadGlobalPatterns() to use global ~/.gitignore?

### DISCUSSION THREAD

**Discussion by romainreignier** [go-git/go-git#793] [Q&A]
In my `~/.gitconfig`, I have the following section:

```
[core]
        excludesfile = /home/user/.gitignore
```

But go-git does not seem to read it with `worktree.Status()` and `status.IsClean()`.

I have found the function [`gitignore.LoadGlobalPatterns()`](https://github.com/go-git/go-git/blob/35f7e6770361a2c16c9b6c44acdc38ae04c75bd3/plumbing/format/gitignore/dir.go#L140) which is supposed to help. But it seems that this function is never called and I don't understand how to use it.

Does someone know how to use it?

Thanks in advance

**Comment by romainreignier** [comment]
With a Github global search, I have found an usage of it:
https://github.com/apache/skywalking-eyes/blob/5dfa68f93380a5e57259faaf95088b7f133b5778/pkg/header/check.go#L84-L94

I should have searched harder before asking. But an example about that use case might be interesting, isn't it?

**Comment by AriehSchneier** [comment]
You should not need to read that file manually, it should already read it for you automatically...
If this doesn't work correctly and you need to do it manually I would suggest leaving this open as a bug.
